hpke/hpke.go

```diff
@@ -0,0 +1,253 @@
+// Package hpke implements the Hybrid Public Key Encryption (HPKE) as specified
+// by draft-irtf-cfrg-hpke-06.
+//
+// HPKE works for any combination of an asymmetric-key encapsulation mechanism
+// (KEM), a key derivation function (KDF), and an authenticated symmetric-key
+// encryption scheme with additional data (AEAD).
+//
+// Specification in
+// https://www.ietf.org/archive/id/draft-irtf-cfrg-hpke-06.html
+package hpke
+
+import (
+	"errors"
+
+	"github.com/cloudflare/circl/kem"
+)
+
+const versionLabel = "HPKE-06"
+
+// Exporter allows exporting secrets from an HPKE context using a
+// variable-length PRF. Export takes as input a context string expCtx and a
+// desired length (in bytes), and produces a secret derived from the internal
+// exporter secret using the corresponding KDF Expand function.
```

Comment:
What are the limits on `len`? Might an implementation panic if `len` is too long? If so, this should be documented here.

Reply:
added some a check, doc, and test